File: CSharp/Fund/Account.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace BankProgram.Fund
{
    public abstract class Account : Bank
    {

        protected String accountOwner;
        protected double accountBalance;

        public Account(String bankName, String owner, double balance) : base()
        {
            BankName = bankName;
            accountBalance = balance;
            accountOwner = owner;

        }
        public Account(String bankName, String owner) : base()
        {
            BankName = bankName;
            accountBalance = 0.0;
            accountOwner = owner;
        }
        public double getAmount {
            get { return accountBalance;
            }
        }

        public virtual void withdraw(double amount)
        {
            if (accountBalance < amount || accountBalance == 0)
            {
                throw new ArithmeticException("Can't withdraw the money from "+ this.GetType().Name);
            }
            else

                accountBalance -= amount;
        }
        public void deposit(double amount)
        {
            accountBalance += amount;
        }
        public void tranfer(double amount, Account toAccount)
        {
            withdraw(amount);
            toAccount.deposit(amount);

        }
        public override string ToString()
        {
            return base.ToString()+  accountOwner+", you have $" + accountBalance+ " in your " + this.GetType().Name + " account" ;
        }



    }
}

File: Java/bank/Account.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package bank;

/**
 *
 * @author samue
 */
public class Account extends Bank  {
    
    protected String accountOwner;
    protected double accountBalance;
    public Account(String bankName,String Owner, double account  )
    {
        super(bankName);
        accountOwner = Owner;
        accountBalance = account;
    }
    public Account(String bankName, String Owner )
    {
        super(bankName);
        accountOwner= Owner;
        accountBalance = 0;
    }
    public  void deposit(double amount)
    {
        accountBalance+=amount;
    }
   public void withdraw (double amount )
   {
       if(accountBalance == 0 || accountBalance < amount )
    System.out.println("Can't withdraw "+amount);
       else 
       accountBalance-= amount;
   }
   // this is for debugging 
   public double getBalance()
   {
       return accountBalance;
   }
   public  void transfer(double amount,  Account toAccount)
   {
       if(accountBalance<amount )
       System.out.println("Can't tranfer the funds");
      
       else{
       withdraw(amount);
       toAccount.deposit(amount);
       }
       
     
   }
   
   
}
